src/lcm/dispatchers.py

```diff
@@ -40,42 +50,53 @@ def spacemap(func, dense_vars, sparse_vars, *, dense_first):
             above but there might be additional dimensions.
 
     """
-    func = allow_args(func)  # vmap cannot deal with keyword-only arguments
-
+    # Check inputs
+    # ==================================================================================
     if not set(dense_vars).isdisjoint(sparse_vars):
         raise ValueError("dense_vars and sparse_vars overlap.")
 
-    _all_variables = dense_vars + sparse_vars
-    if len(_all_variables) != len(set(_all_variables)):
-        raise ValueError("Same argument provided more than once.")
+    if len(set(dense_vars)) < len(dense_vars):
+        raise ValueError("Same argument provided more than once in dense variables.")
+
+    if len(set(sparse_vars)) < len(sparse_vars):
+        raise ValueError("Same argument provided more than once in sparse variables.")
+
+    # Preparations
+    # ==================================================================================
+    func = allow_args(func)  # vmap cannot deal with keyword-only arguments
 
     signature = inspect.signature(func)
     parameters = list(signature.parameters)
 
-    positions = [parameters.index(cv) for cv in sparse_vars]
-    in_axes = []
-    for i in range(len(parameters)):
-        if i in positions:
-            in_axes.append(0)
+    sparse_positions = [parameters.index(cv) for cv in sparse_vars]
+
+    sparse_in_axes = []  # type: list[int | None]
```

Comment:
`in_axes_for_sparse_vars`? With the name change, I wanted to make clear that these `in_axes` are used by a `vmap` call vectorizing the function over the sparse variables.

Reply:
I think I'd prefer `in_axes_for_vmap`, makes it clear where it will be used and there's a 1:1 mapping to `sparse_vars`, which is very clear once looking at the code description there.

Reply:
Not entirely convinced, but I also do not have such a strong opinion on this. My last suggestion would be `in_axes_sparse_vmap`. If you don't like that we can stick with `in_axes_for_vmap`.

Reply:
Either `in_axes_for_sparse_vars` or `in_axes_for_vmap`, the mixture makes it more confusing :laughing:

In the end it depends on how familiar you expect readers of that snippet to be with `vmap`'s interface. I am not, hence I would prefer the latter. But if it is clear to most readers where the name `in_axes` comes from, the former is better. Leaving this up to you!